### ISSUE TITLE
Changeset reference is not processed when preceded by @-mention

### ISSUE THREAD

**Issue by luzpaz** [mantisbt-plugins/source-integration#204]
MantisBT 2,2
Latest source-integration version (GitHub)

Example: https://freecadweb.org/tracker/view.php?id=2954#c8632
 `c:FreeCAD:2dc12cd5416830ee67c2aba362d8285acadfb26f:`
 `d:FreeCAD:2dc12cd5416830ee67c2aba362d8285acadfb26f:`
do not work because the whole SHA was used. As you can see the next part of the thread that it works because it's not referencing the whole SHA `c:FreeCAD:2dc12cd5416830e:`

**Comment by dregad** [comment]
I actually faced the same issue in http://www.mantisbt.org/bugs/view.php?id=17145#c56028 a few days ago.

Following your bug report, I did some additional research, and it would appear that the changeset linking does not work when it follows an @-mention, regardless of whether the whole SHA or a shorter one is used.

**Comment by dregad** [comment]
The problem is that by the time it reaches the plugin's _display_formatted()_ function, @-mention processing has already taken place, so the received text is 

```
<span class="mention"><a class="user" href="https://path/to/mantisbt/view_user_page.php?id=1234">@dregad</a></span> testing start by mention full SHA :mantisbt:ebb54bda7d6a57917639e58322c128ec6eb6b345:<br />
```

The regex effectively starts matching from the @-mention's _href_ (`https://`, beginning with the `s` if https, the `p` if http or from `:` otherwise).

To fix this, the regex, should check that the changeset reference is preceded by whitespace or the beginning of the string.

**Comment by dregad** [comment]
Problem is resolved in _master-1.3.x_ branch. Will merge in master later.

Note that commit 3dbec8114247b930f378dbda248310307a365cbf) applies cleanly on _master_ as well, so you can patch your installation if you need it now.

**Comment by dregad** [comment]
Merged in master f4b67d7de42987e9b6aea1e1f1f6b702447f5f14
